### ISSUE TITLE
FEAT: check if using dot command in posix shell without a slash in the filename

### ISSUE THREAD

**Issue by jumanjiman** [koalaman/shellcheck#1081]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
. path_without_a_slash
```

#### Here's what shellcheck currently says:

SC1091: Not following: path_without_a_slash was not specified as input (see shellcheck -x).

#### Here's what I wanted or expected to see:

SC1091: Not following: path_without_a_slash was not specified as input (see shellcheck -x).
SCxxxx: In POSIX sh, dot command follows PATH if filename lacks a slash

#### POSIX specification

http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#dot

**Comment by pabs3** [comment]
Please note that this syntax works in bash, because it adds the current directory to the path used for the dot command.
